contracts/PriceOracle/PriceOracle.sol

```diff
@@ -110,34 +103,30 @@ contract PriceOracle is Ownable2Step {
         // Initialize the aggregator to read the price from
         AggregatorV3Interface priceFeed = AggregatorV3Interface(config.priceFeed);
         // Retrieve decimals from feed for formatting
-        uint8 decimals = priceFeed.decimals();
-        // Fail safe check since this is an unrealistic scenario. The number of digits that the int256 answer can have naturally limits decimals
-        // but this protects against an erroneous uint8 values
-        if (decimals > MAX_DECIMALS) revert DecimalsTooLarge(decimals);
+        uint8 feedDecimals = priceFeed.decimals();
         // Retrieve price from feed
         (
             /* uint80 roundID */,
             int256 answer,
-            /*uint startedAt*/,
-            /*uint timeStamp*/,
+            /*uint256 startedAt*/,
+            /*uint256 updatedAt*/,
             /*uint80 answeredInRound*/
         ) = priceFeed.latestRoundData();
-        if (answer < 0) revert NegativePrice(answer);
+        // Invalid price returned by feed. Comptroller expects 0 price on error.
+        if (answer <= 0) return 0;
         uint256 price = uint256(answer);
 
         // Number of decimals determines whether the price needs to be multiplied or divided for scaling
         // Handle the 2 scenarios separately to ensure a non-fractional scale value
-        if (decimals <= PRICE_SCALE) {
+        if (feedDecimals + config.underlyingAssetDecimals <= PRICE_SCALE) {
             // Decimals is always >=0 so the scale max value is 1e36 here and not at risk of overflowing
-            uint256 scale = 10 ** (PRICE_SCALE - decimals);
-            return FullMath.mulDiv(price, scale, config.baseUnit);
+            uint256 scale = 10 ** (PRICE_SCALE - feedDecimals - config.underlyingAssetDecimals);
+            return price * scale;
         } else {
-            // Decimals is capped at 72 by earlier validation so scale max value is 1e36 here and not at risk of overflowing
-            uint256 scale = 10 ** (decimals - PRICE_SCALE);
-            // Divide price by scale and base unit separately
-            // Multiplying scale and base unit before dividing price could result in an overflow
-            uint256 scaledPrice = FullMath.mulDiv(price, 1, scale);
-            return FullMath.mulDiv(scaledPrice, 1, config.baseUnit);
+            // Sum of feed and underlying asset decimals is capped at 72 by earlier validation so scale max value is 1e36 here
+            // and not at risk of overflowing
+            uint256 scale = 10 ** (feedDecimals + config.underlyingAssetDecimals - PRICE_SCALE);
+            return price / scale;
```

Comment:
Will we ever unintentionally round down to 0 here?  I.e `price < scale`.

Reply:
I think it's possible but may not be unintentional. To maintain a non-zero value, we wouldn't be able to return a properly formatted answer. If we fail to format, we'd probably want to return an error which is the same as returning 0. I think even the Compound UAV can return a 0 price after formatting. If our feed returned 0.00000001, the `validate` function would have formatted this down to 6 decimal places resulting in 0.